src/Service/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -124,18 +124,8 @@ public abstract class BaseSqlQueryStructure : BaseQueryStructure
         /// </summary>
         public Type GetColumnSystemType(string columnName)
         {
-            if (GetUnderlyingSourceDefinition().Columns.TryGetValue(columnName, out ColumnDefinition? column))
-            {
-                return column.SystemType;
-            }
-            else
-            {
-                throw new DataApiBuilderException(
-                    message: $"{columnName} is not a valid column of {DatabaseObject.Name}",
-                    statusCode: HttpStatusCode.BadRequest,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest
-                    );
-            }
+            GetUnderlyingSourceDefinition().Columns.TryGetValue(columnName, out ColumnDefinition? column);
+            return column!.SystemType;
```

Comment:
>column! [](http://example.com/codeflow?start=19&length=7)

if you've confirmed that there can't be an occurrence of when the exception is raised, the if statement would still be helpful to do null check on column and avoid usage of null forgiving operator.

Reply:
I agree. Retaining the checks.